src/core.rs

```diff
@@ -8,27 +8,19 @@
 
 use byteorder::{ByteOrder, LittleEndian};
 
-/// Error that occured while performing and I2C Operation
-#[derive(Debug)]
-pub enum I2CError {
-    NotSupported,
-    Other(&'static str),
-}
-
-/// Result of an I2C Operation
-pub type I2CResult<T> = Result<T, I2CError>;
-
 /// Interface to an I2C Slave Device from an I2C Master
 ///
 /// Typical implementations will store state with references to the bus
 /// in use and the address of the slave device.  The trait is based on the
 /// Linux i2cdev interface.
 pub trait I2CDevice {
+    type Error;
```

Comment:
I tried requiring `std::error::Error` but then found out `nix::Error` didn't impl it. I plan on addressing that in its own PR. `fmt::Debug` wouldn't be too bad to require, since `Result::unwrap` needs it.